checks/linux.py

```diff
@@ -0,0 +1,176 @@
+
+    hasNoPipewireCapture = search('[pipewire] No capture', lines)
+    if len(hasNoPipewireCapture) > 0:
+        return [LEVEL_CRITICAL, "No PipeWire capture on Wayland",
+                """In order to capture displays or windows under Wayland, OBS requires the appropriate PipeWire capture portal for your Desktop Environment.<br><br>
+                An overview of available capture portals can be found on the Arch Linux Wiki:<br>
+                <a href='https://wiki.archlinux.org/title/XDG_Desktop_Portal'>XDG Desktop Portal</a><br>
+                Note that the availability of Window and/or Display capture depends on your Desktop Environment's implementation of these portals."""]
+
+    return [LEVEL_INFO, "Wayland",
+            """Window and Display Captures are available via <a href='https://wiki.archlinux.org/title/XDG_Desktop_Portal'>XDG Desktop Portal</a><br>.
+            Please note that the availability of captures and specific features depends on your Desktop Environment's implementation of these portals.<br><br>
+            Global Keyboard Shortcuts are not currently available under Wayland."""]
+
+
+def checkX11(lines):
+    isDistroNix = search('Distribution:', lines)
+    isFlatpak = search('Flatpak Runtime:', lines)
+
+    if (len(isDistroNix) <= 0) and (len(isFlatpak) <= 0):
+        return
+
+    sessionTypeLine = getSessionTypeLine(lines)
+    if not sessionTypeLine:
+        return
+
+    sessionType = sessionTypeLine.split()[3]
+    if sessionType != 'x11':
+        return
+
+    # obsolete PW sources
+    hasPipewireCaptureDesktop = search('pipewire-desktop-capture-source', lines)
+    hasPipewireCaptureWindow = search('pipewire-window-capture-source', lines)
+    # unified PW source
+    hasPipewireCaptureScreen = search('pipewire-screen-capture-source', lines)
+
+    if (len(hasPipewireCaptureDesktop) > 0) or (len(hasPipewireCaptureWindow) > 0) or (len(hasPipewireCaptureScreen) > 0):
+        return [LEVEL_WARNING, "PipeWire capture on X11",
+                """While technically possible, most Desktop Environments do not implement the PipeWire capture portals on X11. Your captures will therefore likely not work, i.e. you will be unable to pick a window or display, or the selected source will stay empty.<br><br>
+                We generally recommend using \"Window Capture (Xcomposite)\" on X11, as \"Display Capture (XSHM)\" can introduce bottlenecks depending on your setup.<br><br>
+                From a technical standpoint a Desktop Environment should not advertise the availability of these captures if they are not implemented. This behaviour can therefore be considered a bug in your Desktop Environment."""]
+
+    return [LEVEL_INFO, "X11",
+            "Window Capture is available via Xcomposite, while Display Capture is available via XSHM. We generally recommend sticking to \"Window Capture (Xcomposite)\" since \"Display Capture (XSHM)\" can introduce bottlenecks depending on your setup."]
```

Comment:
Should this be emitted all the time even if the user is not using any captures?

Reply:
Since this is only `INFO` this was intended, yes. We often get questions of what capture method to use, and there's not always someone around.

Reply:
I have mixed feelings about always emitting an INFO like this, but I don't have enough experience in Linux support to accurately weigh the benefits vs. the noise.

Reply:
If it helps, it's not much different then Windows logs always showing the Windows version and Admin status. It's a helper info for Supporters (although not being active in w-s I don't know how much it helps or what the differences are), and for endusers reading the analysis it doesn't provide much detail either.

The big difference being that on Linux there's more variables to consider (Distro, Desktop Environment, and Window System mostly). So yes we end up with 3 info bulletins instead of 2, but it is information we look at in every log anyway. So it takes a click off supporters and if noone's around it might help the occasional user. If not, well then they'll have to wait for someone to come around anyway 😅

Reply:
Always showing the Windows version is "fine", which is why I'm okay with showing the Distro info elsewhere for Linux (though it might be nice if they could be a single item "system info"). This is mostly because we're aware of issues that only exist or are fixed in specific OS versions that are otherwise hard to diagnose from logged info.

The point I'm making here is that we're emitting an INFO giving blanket capture advice whether or not any such sources are detected. For all we know, the user won't be using any of the captures mentioned (maybe they're only using v4l2 or teleport or browser).

Reply:
Yeah I get what you're saying.
I just honestly don't really see it being problematic.
If a user doesn't want to capture either windows or a display, they can just ignore it since it's clearly about window and display capture. Maybe we could reword it a little:

> If you wish to capture a window or an entire display, captures are available via Xcomposite and XSHM respectively. We generally recommend sticking to "Window Capture (Xcomposite)" since "Display Capture (XSHM)" can introduce bottlenecks depending on your setup.

That should make it even clearer that it can and should be ignored if no captures are intended.

If users do end up getting confused by it we can remove it anytime as well. We won't know unless we have it first though.

And going back to the windows logs, the same can be said about the "Not Admin" factoid, which can lead users to assume they should be running it as admin (similar to the audio buffering factoid in the other PR).